File: problems/2816/Lc2816test.py
from listnode import ListNode
from Lc2816 import Solution
# from Lc2816_list import Solution


def main():
    sol = Solution()

    head = ListNode.fromList([1, 8, 9])
    result = sol.doubleIt(head).toList()
    expected = [3, 7, 8]
    assert expected == result, f'expected {expected}, but {result}'

    head = ListNode.fromList([9, 9, 9])
    result = sol.doubleIt(head).toList()
    expected = [1, 9, 9, 8]
    assert expected == result, f'expected {expected}, but {result}'


if __name__ == '__main__':
    main()

File: problems/2816/Lc2816test.ts
import { ListNode, array2ListNode, ListNode2array } from './listnode';
import { doubleIt } from './Lc2816';
// import { doubleIt } from './Lc2816_array';

function main() {
  let result: number[], expected: number[], head: ListNode | null;

  head = array2ListNode([1, 8, 9]);
  result = ListNode2array(doubleIt(head));
  expected = [3, 7, 8];
  if (!(expected.length === result.length
    && expected.every((val, idx) => (val === result[idx]))))
    throw new Error(`expected ${expected}, but ${result}`);

  head = array2ListNode([9, 9, 9]);
  result = ListNode2array(doubleIt(head));
  expected = [1, 9, 9, 8];
  if (!(expected.length === result.length
    && expected.every((val, idx) => (val === result[idx]))))
    throw new Error(`expected ${expected}, but ${result}`);
};
main();
